torchgeo/datasets/worldstrat.py

```diff
@@ -0,0 +1,358 @@
+        )
+
+        return sample
+
+    def _load_sentinel_data(self, data_dir: str) -> Tensor:
+        """Load Sentinel data for a given AOI in a data directory.
+
+        Args:
+            data_dir: Directory containing the Sentinel data, in the dataset
+                this is either the L1C or L2A directory with time-series.
+
+        Returns:
+            Loaded Sentinel data stacked as tensor so [T, C, H, W].
+        """
+        tiff_paths = glob(
+            os.path.join(data_dir, f'*{os.path.basename(data_dir)}_data.tiff'),
+            recursive=True,
+        )
+
+        # load and stack the data
+        data = []
+        for tiff_path in tiff_paths:
+            data.append(self._load_tiff(tiff_path))
+
+        return torch.stack(data).float()
+
+    def _load_tiff(self, tiff_path: str) -> Tensor:
+        """Load a tiff file as a tensor."""
+        with rasterio.open(tiff_path) as src:
+            data = src.read()
+            tensor = array_to_tensor(data)
+        return tensor
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset."""
+        return len(self.file_path_df)
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # check if directories are present
+        exists = []
+        split_info_path = os.path.join(
+            self.root, self.file_info_dict['train_val_test_split']['filename']
+        )
+        if os.path.exists(split_info_path):
+            df = pd.read_csv(split_info_path)
+            df = df[df['split'] == self.split]
+            # check that all tiles are present
+            for tile in df['tile']:
+                exists.append(os.path.exists(os.path.join(self.root, tile)))
+        else:
+            exists.append(False)
+
+        if all(exists):
+            return
+
+        # check if downloaded files are present
+        exists = []
+        for file in self.file_info_dict.values():
+            path = os.path.join(self.root, file['filename'])
+            if os.path.exists(path):
+                if self.checksum:
+                    md5 = file['md5']
+                    if not check_integrity(path, md5):
+                        raise RuntimeError(f'Archive {file["filename"]} corrupted')
+                exists.append(True)
+            else:
+                exists.append(False)
+
+        if all(exists):
+            # extract files
+            self._extract()
+            return
+
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        # download
+        self._download()
+
+    def _extract(self) -> None:
+        """Extract tar balls to root directory."""
+        for file in self.file_info_dict.values():
+            if 'tar.gz' in file['filename']:
+                extract_archive(os.path.join(self.root, file['filename']), self.root)
+
+    def _download(self) -> None:
+        """Download the dataset and extract it."""
+        # TODO: implement download
```

Comment:
Is this done now?

Reply:
Download works, just waiting for them to fix the dataset, which will probably (hopefully) just result in a new download link